xrpl/wallet/main.py

```diff
@@ -16,69 +17,141 @@ class Wallet:
         a hexadecimal string.
         """
 
-        self.private_key = sk
+        self.private_key = private_key
         """
         The private key that is used to create signatures, as a hexadecimal
         string. MUST be kept secret!
         """
 
-        self.classic_address = derive_classic_address(self.public_key)
-        """The address that publicly identifies this wallet, as a base58 string."""
+        self._classic_address = (
+            ensure_classic_address(master_address)
+            if master_address is not None
+            else derive_classic_address(self.public_key)
+        )
+        """Internal variable for classic_address. Use classic_address instead."""
 
-        self.sequence = sequence
+        self.seed = seed
         """
-        The next available sequence number to use for transactions from this
-        wallet.
-        Must be updated by the user. Increments on the ledger with every successful
-        transaction submission, and stays the same with every failed transaction
-        submission.
+        The core value that is used to derive all other information about
+        this wallet. MUST be kept secret!
         """
 
     @classmethod
     def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
+        cls: Type[Wallet], algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
     ) -> Wallet:
         """
         Generates a new seed and Wallet.
 
         Args:
-            crypto_algorithm: The key-generation algorithm to use when generating the
-                seed. The default is Ed25519.
+            algorithm: The key-generation algorithm to use when generating the seed.
+                The default is `ED25519`.
 
         Returns:
             The wallet that is generated from the given seed.
         """
-        seed = generate_seed(algorithm=crypto_algorithm)
-        return cls(seed, sequence=0, algorithm=crypto_algorithm)
+        seed = generate_seed(algorithm=algorithm)
+        return Wallet.from_seed(seed, algorithm=algorithm)
+
+    @classmethod
+    def from_seed(
+        cls: Type[Wallet],
+        seed: str,
+        *,
+        master_address: Optional[str] = None,
+        algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519,
+    ) -> Wallet:
+        """
+        Generates a new Wallet from seed (secret).
+
+        Args:
+            seed: The seed (secret) used to derive the account keys.
+            master_address: Include if a Wallet uses a Regular Key Pair. This sets the
+                address that this wallet corresponds to. The default is `None`.
+            algorithm: The key-generation algorithm to use when generating the seed.
+                The default is `ED25519`.
+
+        Returns:
+            The wallet that is generated from the given secret.
+        """
+        public_key, private_key = derive_keypair(seed, algorithm=algorithm)
```

Comment:
I don't know if we should be setting a default algorithm. Decoding based on the seed prefix seems to make sense to me as better behavior that users can override if they have a problem with it. (Rather than requiring every secp256k1 user to override)

Reply:
We can't decode based on the seed prefix if they don't use a `sEd` key - an `s` key could be either algorithm.

Reply:
@mvadari How does decode_seed work right now if it is trying to figure out which algorithm to use? From the code, it seems like it just tries prefixes and see if it comes out to a valid outcome. For an `s` key, does this usually default to a ED25519 algorithm based on order of algorithms it tries?

Reply:
We misunderstood how seeds work when it was originally written - we thought that ED25519 keys had to start with `sEd` and secp256k1 keys just started with `s`. It turns out that this is not correct - ED25519 keys can be encoded with `sEd`, but they can also be encoded with just `s`. The algorithm is only really used to get the keys from the seed; the seed doesn't need to be encoded in any specific way. 

For example, when you call `wallet_propose` on rippled and request an ED25519 key, this is what you get:
{'account_id': 'r92k4fsv4ehYk3GzFRL1gigRk2aie8ZRkR',
 'key_type': 'ed25519',
 'master_key': 'BAIT RULE LIMB WARN IONS BAIL GAUR CORK COMB VOID SIT MARS',
 'master_seed': 'shEdVvJUB7a3eq7TTFVxfLpjEwKb7',
 'master_seed_hex': '6FC9A37AA969ED89A1ECD97BBE821A51',
 'public_key': 'aKNq5Z4GNoKc9gpxUiJkiuNjBjRf7PvhcQ6weAwCtcjZQodjviXT',
 'public_key_hex': 'ED09B678EFB1B02B8C3372FC1B31E0562508983DAB9F4E53464A51F47E93D5C4F8'}

So the somewhathacky way that it's written right now (to maintain backwards compatibility) is basically this:
```py
if algorithm is not None:
  if key.startswith(`sEd`) and algorithm == 'secp256k1':
    raise Exception()
  # encode with the algorithm provided
if key.startswith(`sEd`):
  # use ed25519
else:
  # use secp256k1
```

Reply:
@mvadari So for future, are you fine with just defaulting everything to ed25519 unless otherwise specified?

Reply:
Yes, I think that would be reasonable - switch the logic to default to ed25519 unless otherwise specified. But this has to be really well-documented, because it'll be difficult to debug.